compiler/rustc_infer/src/infer/higher_ranked/mod.rs

```diff
@@ -24,9 +24,9 @@ impl<'a, 'tcx> CombineFields<'a, 'tcx> {
         // as-is, we need to do some extra work here in order to make sure
         // that function subtyping works correctly with respect to regions
         //
-        // Note: this is a subtle algorithm.  For a full explanation,
-        // please see the large comment at the end of the file in the (inlined) module
-        // `doc`.
+        // Note: this is a subtle algorithm.  For a full explanation, please see
+        // the rustc book:
+        // https://rustc-dev-guide.rust-lang.org/traits/hrtb.html
```

Comment:
I think https://rustc-dev-guide.rust-lang.org/borrow_check/region_inference/placeholders_and_universes.html might be a better link, it's the one shown in https://rustc-dev-guide.rust-lang.org/appendix/glossary.html?highlight=skolem#glossary for 'skolemization', which is what the comment originally said when introduced in 23652efffb32de3f2babcb6204f49268b19cc0e6.
```suggestion
        // <https://rustc-dev-guide.rust-lang.org/borrow_check/region_inference/placeholders_and_universes.html>
```